### PR TITLE
disabled firebase crashlytics

### PR DESCRIPTION
**Pull request by suyash01**
## Pull Request (PR) Checklist
Please check if your pull request fulfills the following requirements:
- [x] The PR is submitted to the `main` branch.
- [x] I've read the [Contribution Guidelines](https://github.com/Ivy-Apps/ivy-wallet/blob/main/CONTRIBUTING.md) and my PR doesn't break the rules.
- [x] I've read and understand the [Developer Guidelines](https://github.com/Ivy-Apps/ivy-wallet/blob/main/docs/Guidelines.md).
- [ ] I confirm that I've run the code locally and everything works as expected.
- [ ] 🎬 I've attached a **screen recoding** of the changes.
- [x] My PR includes only the necessary changes to fix the issue (i.e., no unnecessary files or lines of code are changed).

> Tip: drag & drop the video to the PR description.

## What's changed?

Describe with a few bullets **what's new:**
- a
- b
- c

> 💡 Tip: Please, attach screenshots and screen recordings. It helps a lot!

## Risk Factors

**What may go wrong if we merge your PR?**

- a
- b

**In what cases your code won't work?**

- a
- b

## Does this PR closes any GitHub Issues?

Check **[Ivy Wallet Issues](https://github.com/Ivy-Apps/ivy-wallet/issues)**.

- Closes #ISSUE_NUMBER

> Replace `ISSUE_NUMBER` with the number of the GitHub issue that this PR is fixing. If you've done that correctly, you'll see the issue title linked when previewing your PR description.

> **!Note:** Do **not** link the PR number. Link the number/id of the GitHub Issue from [issues](https://github.com/Ivy-Apps/ivy-wallet/issues).

## Troubleshooting CI failures ❌

GitHub Actions failing? Read our [CI Troubleshooting guide](https://github.com/Ivy-Apps/ivy-wallet/blob/main/docs/CI-Troubleshooting.md).

### CODE REVIEW

**Comment by suyash01** [comment]
@ILIYANGERMANOV I have made some changes but from the logs it is not entirely clear if the crash logs are being pushed to firebase or not. If you get some time, could you test this with firebase? I have made the transaction button to throw exception to emulate crash.

File: app/build.gradle.kts [hunk 1]
```diff
@@ -70,6 +70,11 @@ android {
 
             applicationIdSuffix = ".debug"
             resValue("string", "app_name", "Ivy Wallet Debug")
+
+            firebaseCrashlytics {
+                mappingFileUploadEnabled = false
```
Comment (ILIYANGERMANOV) [inline app/build.gradle.kts:75]: This removes the deobfuscation mapping. I.e., still uploads crashes but as gibberish.

**Review by ILIYANGERMANOV — CHANGES_REQUESTED** [review]
To effectively remove Firebase you need to somehow declare it as "release" only dependencies. If the library is in the dependencies, we gain no benefits.

The problem that we need to solve is:
- get all Firebase and Google dependencies removed from the classpath on "debug" builds
- keep Firebase only for "release" builds

File: app/src/main/java/com/ivy/wallet/IvyAndroidApp.kt [hunk 2]
```diff
@@ -27,7 +29,10 @@ class IvyAndroidApp : Application(), Configuration.Provider {
         appContext = this
 
         if (BuildConfig.DEBUG) {
+            Firebase.crashlytics.setCrashlyticsCollectionEnabled(false)
             Timber.plant(DebugTree())
+        } else {
+            Firebase.crashlytics.setCrashlyticsCollectionEnabled(true)
```
Comment (ILIYANGERMANOV) [inline app/src/main/java/com/ivy/wallet/IvyAndroidApp.kt:35]: This should probably work, especially if you've disable Firebase auto-init

**Comment by suyash01** [comment]
@ILIYANGERMANOV I am spending way too much time on this and my lack of firebase knowledge is not helping. If you have any idea let know otherwise i will close this PR and unassign myself from the ticket.

**Comment by ILIYANGERMANOV** [comment]
> @ILIYANGERMANOV I am spending way too much time on this and my lack of firebase knowledge is not helping. If you have any idea let know otherwise i will close this PR and unassign myself from the ticket.

Yeah, it's a tricky one. Feel free to unassign^

**Pull request closed without merging**